service/resource/cloudformation/adapter.go

```diff
@@ -0,0 +1,81 @@
+package cloudformation
+
+import (
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/microerror"
+
+	awsutil "github.com/giantswarm/aws-operator/client/aws"
+)
+
+type hydrater func(awstpr.CustomObject, awsutil.Clients) error
+
+type adapter struct {
+	ASGType string
+
+	lauchConfigAdapter
+	autoScalingGroupAdapter
+}
+
+type lauchConfigAdapter struct {
+	ImageID                  string
+	SecurityGroupID          string
+	InstanceType             string
+	IAMInstanceProfileName   string
+	BlockDeviceMappings      []BlockDeviceMapping
+	AssociatePublicIPAddress bool
+	SmallCloudConfig         string
+}
+
+type BlockDeviceMapping struct {
+	DeviceName string
+	VolumeSize string
+	VolumeType string
+}
+
+type autoScalingGroupAdapter struct {
+	SubnetID               string
+	AZ                     string
+	ASGMinSize             int
+	ASGMaxSize             int
+	LoadBalancerName       string
+	HealthCheckGracePeriod string
+	MinInstancesInService  int
+	MaxBatchSize           string
+	RollingUpdatePauseTime string
+}
+
+func newAdapter(customObject awstpr.CustomObject, clients awsutil.Clients) (adapter, error) {
+	a := adapter{}
+
+	a.ASGType = "worker"
+
+	hydraters := []hydrater{
+		a.getAutoScalingGroup,
+		a.getLaunchConfiguration,
+	}
+
+	for _, h := range hydraters {
+		if err := h(customObject, clients); err != nil {
+			return adapter{}, microerror.Mask(err)
+		}
+	}
+
+	return a, nil
+}
+
+func (a *adapter) getLaunchConfiguration(customObject awstpr.CustomObject, clients awsutil.Clients) error {
+	if len(customObject.Spec.AWS.Workers) == 0 {
+		return microerror.Mask(invalidConfigError)
+	}
+
+	a.ImageID = customObject.Spec.AWS.Workers[0].ImageID
+	a.InstanceType = customObject.Spec.AWS.Workers[0].InstanceType
+
+	return nil
+}
+
+func (a *adapter) getAutoScalingGroup(customObject awstpr.CustomObject, clients awsutil.Clients) error {
+	a.AZ = customObject.Spec.AWS.Region
```

Comment:
This should be AZ rather than Region. AFAIK Region is just used for creating the AWS client.

At the moment we only support a single AZ but supporting multiple AZs is something we want to support later.

Reply:
yep, fixed :)